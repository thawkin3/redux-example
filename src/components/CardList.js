import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import Card from './Card';
import './CardList.css';

export const CardList = ({ cards }) => (
    <div className="cardListContainer">
        {cards.map(card => <Card key={card.id} {...card} />)}
    </div>
);

function mapStateToProps({ users }) {
    return { cards: users.cards };
}

CardList.displayName = 'CardList';

export default connect(mapStateToProps, actions)(CardList);
