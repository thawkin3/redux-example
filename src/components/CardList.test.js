import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { CardList } from './CardList';

let wrapper, props;

describe('CardList', () => {
    beforeEach(() => {
        props = {
            cards: [
                {
                    avatar_url: 'url/here/img.jpg',
                    html_url: 'https://github.com/thawkin3',
                    name: 'Tyler Hawkins',
                    company: '@instructure',
                    id: 1,
                },
                {
                    avatar_url: 'url/here/img3.jpg',
                    html_url: 'https://github.com/testPerson',
                    name: 'Test Person',
                    company: '@fakecompany',
                    id: 2,
                },
            ]
        };
        wrapper = shallow(<CardList {...props} />);
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CardList {...props} />, div);
    });

    it('matches the snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders only one CardList', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('renders an element with the class cardListContainer', () => {
        expect(wrapper.find('.cardListContainer').length).toEqual(1);
    });

    it('renders a Card component for every element in the cards array', () => {
        expect(wrapper.find('Card').length).toEqual(2);
    });
});
