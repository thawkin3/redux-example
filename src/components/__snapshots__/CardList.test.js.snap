// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList matches the snapshot 1`] = `
<div
  className="cardListContainer"
>
  <Card
    avatar_url="url/here/img.jpg"
    company="@instructure"
    html_url="https://github.com/thawkin3"
    id={1}
    key="1"
    name="Tyler Hawkins"
  />
  <Card
    avatar_url="url/here/img3.jpg"
    company="@fakecompany"
    html_url="https://github.com/testPerson"
    id={2}
    key="2"
    name="Test Person"
  />
</div>
`;
